core_data_modules/cleaners/somali/demographic_cleaner.py

```diff
@@ -3,8 +3,18 @@
 
 
 class DemographicCleaner(object):
-    @staticmethod
-    def is_noise(text):
+    @classmethod
+    def is_noise(cls, text):
```

Comment:
I'm a little uncomfortable about the name - this is more detecting 'is this a possible demographic string', not noise independent of context. Any thoughts?

Reply:
Consider 'is_possible_demographic_answer'

Reply:
It would actually be `is_not_possible_demographic_answer`

Reply:
Hmm. The problem is the noise regex is still looking for specific patterns known to be noise. The changes here are to reduce the number of occasions where that filter matches demographics.

Reply:
Maybe we should break this into two functions?

Reply:
Although then we need to encourage users to call both, which doesn't feel great

Reply:
Should we split into two functions 'pre_filter_...' and 'is_noise_by_regex' and then have a 'is_noise' method that calls one and then the other?

Reply:
Will keep the name, but re-structure to make it clear that this is a two-phase process